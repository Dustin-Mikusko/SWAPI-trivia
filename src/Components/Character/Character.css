.character-card {
  margin: 25px;
  background-color: white;
  width: 320px;
  height: 300px;
  color: #000;
  transition: all .3s ease-in-out;
}

.character-name {
  font-family: 'Cormorant Garamond', serif;
  color: #000;
}

article:hover {
  transform: scale(1.03);
  box-shadow: 2px 2px 30px 15px rgba(255, 0, 0, 0.776);
}
