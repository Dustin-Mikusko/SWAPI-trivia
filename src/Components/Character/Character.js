import React from 'react';
import './Character.css';

const Character = ({ name, world, population, species, relatedFilms }) => {
  return (
    <section className="character-cards">
      <article className="character-card">
        <h1 className="character-name">{name}</h1>
        <p><span className="bold">World:</span> {world}</p>
        <p><span className="bold">Population:</span> {population}</p>
        <p><span className="bold">Species:</span> {species}</p>
        <p><span className="bold">Related Films:</span> {relatedFilms}</p>
      </article>
    </section>
  )
}

export default Character;
