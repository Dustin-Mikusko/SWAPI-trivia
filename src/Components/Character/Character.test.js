import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';


describe('Character', () => {
  const mockCharacter = {
    name: 'Tom Brady',
    world: 'CaliSaturn',
    population: '1',
    species: "Reeses Pieces Species",
    relatedFilms: ["One", "Two", "Three"]
  }

  const wrapper = shallow(<Character 
    name={mockCharacter.name}
    world={mockCharacter.world}
    population={mockCharacter.population}
    species={mockCharacter.species}
    relatedFilms={mockCharacter.relatedFilms}
  />);
  
  it('should render component to match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });
})
