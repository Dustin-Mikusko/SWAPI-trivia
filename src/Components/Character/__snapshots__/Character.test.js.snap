// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character should render component to match snapshot 1`] = `
<section
  className="character-cards"
>
  <article
    className="character-card"
  >
    <h1
      className="character-name"
    >
      Tom Brady
    </h1>
    <p>
      <span
        className="bold"
      >
        World:
      </span>
       
      CaliSaturn
    </p>
    <p>
      <span
        className="bold"
      >
        Population:
      </span>
       
      1
    </p>
    <p>
      <span
        className="bold"
      >
        Species:
      </span>
       
      Reeses Pieces Species
    </p>
    <p>
      <span
        className="bold"
      >
        Related Films:
      </span>
       
      One
      Two
      Three
    </p>
  </article>
</section>
`;
